src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -120,6 +121,8 @@ public Http2Connection(HttpConnectionPool pool, Stream stream)
 
         private object SyncObject => _httpStreams;
 
+        public bool CanAddNewStream => _canAddNewStream;
```

Comment:
Instead of adding a new piece of state here (_canAddNewStream), couldn't we just have something like IsCreditAvailable on the credit manager and use that? Doesn't seem like a big deal either way, but in general having less state to get out of sync with other state seems good.

Reply:
Yeah, in this approach we can rely on CreditManager since we anyways assign connection to request upfront. `_canAddNewStream` was removed.